nsxt/util/utils.go

```diff
@@ -29,3 +32,23 @@ func NsxVersionHigherOrEqual(ver string) bool {
 	}
 	return currentVersion.Compare(requestedVersion) >= 0
 }
+
+func hash(s string) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(s))
+	return h.Sum32()
+}
+
+func GetVerifyableID(id, extra string) string {
```

Comment:
Do we need the `extra` parameter here? Maybe just randomize id within this function and call it `getRandomVerifiableID`?

Reply:
As I validate these ids already, it makes sense that the id is from the correct resource (so they won't depend on the `nsxt_upgrade_prepare_ready` DS where they should depend on `nsxt_upgrade_prepare` resource etc. This process has many components, it's better to enforce the order as much as possible IMO.